main/CommandHandler.cpp

```diff
@@ -990,6 +990,118 @@ int setAnalogWrite(const uint8_t command[], uint8_t response[])
 }
 
 
+#include "esp_wpa2.h"
+
+int wpa2EntSetIdentity(const uint8_t command[], uint8_t response[]) {
+  char identity[32 + 1];
+
+  memset(identity, 0x00, sizeof(identity));
+  memcpy(identity, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_identity((const unsigned char*)identity, sizeof(identity));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetUsername(const uint8_t command[], uint8_t response[]) {
+  char username[32 + 1];
+
+  memset(username, 0x00, sizeof(username));
+  memcpy(username, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_username((const unsigned char*)username, sizeof(username));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetPassword(const uint8_t command[], uint8_t response[]) {
+  char password[32 + 1];
+
+  memset(password, 0x00, sizeof(password));
+  memcpy(password, &command[4], command[3]);
+
+  esp_wifi_sta_wpa2_ent_set_password((const unsigned char*)password, sizeof(password));
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetCACert(const uint8_t command[], uint8_t response[]) {
+  // use a previously saved file
+  const char* filename = "/fs/ca.pem";
+  FILE* ca_pem = fopen(filename, "rb");
+  if (ca_pem != NULL) {
+    struct stat st;
+    stat(filename, &st);
+    uint8_t* ca_cert_buf = (uint8_t*)malloc(st.st_size);
+    fread(ca_cert_buf, st.st_size, 1, ca_pem);
+    //never deallocate this object since it will be used later
+    esp_wifi_sta_wpa2_ent_set_ca_cert(ca_cert_buf, st.st_size);
+  }
+
+  response[2] = 1; // number of parameters
+  response[3] = 1; // parameter 1 length
+  response[4] = 1;
+
+  return 6;
+}
+
+int wpa2EntSetCertKey(const uint8_t command[], uint8_t response[]) {
+  // use a previously saved file
+  const char* key_filename = "/fs/client.key";
+  FILE* client_key = fopen(key_filename, "rb");
+
+  const char* crt_filename = "/fs/client.crt";
+  FILE* client_crt = fopen(crt_filename, "rb");
+
+  if (client_key != NULL && client_crt != NULL) {
+    struct stat st;
+    stat(key_filename, &st);
+    size_t client_key_buf_size = st.st_size;
+    uint8_t* client_key_buf = (uint8_t*)malloc(st.st_size);
```

Comment:
Since this is stored in RAM anyways, do we need to store it in a file?

Also, should the buffer be freed if the API is called twice in a row?